### PR TITLE
[LDAP] Filter user groups obtained by memberof

### LINKED ISSUE #17119: All LDAP group memberships show up on user's personal page even if group not configured on admin page
### Steps to reproduce

1. Install 8.1 
2. Configure ldap with more than one group
3. Log in as LDAP user and go to personal page
4. Go to Peronsal info

### Expected behaviour

Only see groups that are configured in ldap configuration

### Actual behaviour

See all LDAP groups that the user is a member of even if the groups haven't been configured on the admin page.  Of the groups in this list only aggregate-cross-media-channels has been configured. 

p363

### Server Host info:

Ubuntu 14.04
PHP 5.5.9

### Server configuration:

ownCloud Enterprise Edition 8.1 RC1 (daily) Build:2015-06-23T03:09:53+00:00 25581c7b638a5349c0180b6210b2e1d332e19e49

### PR DESCRIPTION
**Pull request by blizzz**
Fixes #17119 

Please see the original issue for reproduction steps.

This includes an integration test, for information how to set up and run please read the included readme.md (It's not extraordinarily shiny, but does the job for now).

The test's output without the fix:

```
$ sudo ./run-test.sh lib/IntegrationTestAccessGroupsMatchFilter.php 
578e03d0bf173840a46546ebec69da3839a9996111ecb058bf7a028e34d9c121
f8ef11a31cfbb7280efd10a920d3baddb28f90765d4368eafc827f9138397596
CONTAINER ID        IMAGE                       COMMAND             CREATED                  STATUS                  PORTS                           NAMES
f8ef11a31cfb        osixia/phpldapadmin:0.5.1   "/sbin/my_init"     Less than a second ago   Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin   
578e03d0bf17        nickstenning/slapd:latest   "/sbin/my_init"     1 seconds ago            Up Less than a second   127.0.0.1:7770->389/tcp         docker-slapd          

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.17.0.68)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Ialec
created group : RedGroup
created group : BlueGroup
created group : GreenGroup
created group : PurpleGroup
running case1 
running case2 

>>> !!! Test case2 FAILED !!! <<<

Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
```

And with the fix:

```
$ sudo ./run-test.sh lib/IntegrationTestAccessGroupsMatchFilter.php 
e0e54bf49f588f7a2574a4a7a6d77dee5d4b44390aa8b41a3d0234b6c3d176c7
8e8cf0b5e4432260471ae98fe66ac7e888c400b623f4716bfb44f063f1ecd5be
CONTAINER ID        IMAGE                       COMMAND             CREATED                  STATUS                  PORTS                           NAMES
8e8cf0b5e443        osixia/phpldapadmin:0.5.1   "/sbin/my_init"     Less than a second ago   Up Less than a second   80/tcp, 0.0.0.0:8443->443/tcp   docker-phpldapadmin   
e0e54bf49f58        nickstenning/slapd:latest   "/sbin/my_init"     Less than a second ago   Up Less than a second   127.0.0.1:7770->389/tcp         docker-slapd          

LDAP server now available under 127.0.0.1:7770 (internal IP is 172.17.0.70)
phpldapadmin now available under https://127.0.0.1:8443

created user : Alice Ilcea
created group : RedGroup
created group : BlueGroup
created group : GreenGroup
created group : PurpleGroup
running case1 
running case2 
Tests succeeded
Stopping and resetting containers
docker-slapd
docker-phpldapadmin
docker-slapd
docker-phpldapadmin
```

Kudos for @butonic @MorrisJobke and @PVince81 for https://github.com/owncloud/administration/tree/master/ldap-testing which is used for it.

Please test (also users page with plentiful of users to see possible performance impact) @jnfrmarks @jvillafanez @davitol

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Awesome tests - thx @blizzz - we need to think about possibilities to integrate this into our ci mechanisms.

**Comment by DeepDiver1975** [comment]
@blizzz 

````
OCA\user_ldap\tests\Test_Group_Ldap::testGetUserGroupsMemberOf
Invalid argument supplied for foreach()

/var/jenkins/workspace/pull-request-analyser-ng-simple@3/label/SLAVE/apps/user_ldap/group_ldap.php:384
/var/jenkins/workspace/pull-request-analyser-ng-simple@3/label/SLAVE/apps/user_ldap/tests/group_ldap.php:408

File: apps/user_ldap/tests/integration/setup-scripts/createExplicitGroups.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+include __DIR__ . '/config.php';
```
Comment (LukasReschke) [inline apps/user_ldap/tests/integration/setup-scripts/createExplicitGroups.php]: Just for the sake of making me happier is it possible to:

1. Add a `.htaccess` file to the tests folder
2. Add a check if this file is accessed via CLI 

Thanks :smile:

**Comment by blizzz** [comment]
@LukasReschke addressed

**Comment by LukasReschke** [comment]
:rocket:

**Comment by jnfrmarks** [comment]
@blizzz 

I can test this - is there a way to get a patch so I can update my installation?

**Comment by blizzz** [comment]
@jnfrmarks what about https://patch-diff.githubusercontent.com/raw/owncloud/core/pull/17255.diff ?

**Comment by jnfrmarks** [comment]
:+1: 

Validated with two openLDAP servers configured, with one group > 700 users.

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by blizzz** [comment]
@karlitschek backport request to stable8.1

**Comment by karlitschek** [comment]
and there is begins :-)
Please backport :+1:

**Comment by blizzz** [comment]
Hehe ;) It was already merged only a bit before you branched off, no action necessary. I edited my last post, but Github doesn't send out info on edits, unfortunately.

**Comment by blizzz** [comment]
As I now must ascertain, it somehow was not part when branching off 8.1 :speak_no_evil: :angry:

**Comment by blizzz** [comment]
Backport now here https://github.com/owncloud/core/pull/17704

**Pull request merged**